fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupClassifier.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    public void setQueryTypes(Set<QueryType> queryTypes) {
+        this.queryTypes = queryTypes;
+    }
+
+    public String getSourceIp() {
+        return sourceIp;
+    }
+
+    public void setSourceIp(String sourceIp) {
+        this.sourceIp = sourceIp;
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        String json = GsonUtils.GSON.toJson(this);
+        Text.writeString(out, json);
+    }
+
+    public boolean isSatisfied(String user, String role, QueryType queryType, String sourceIp) {
+        if (!isVisible(user, role, sourceIp)) {
+            return false;
+        }
+        if (this.queryTypes != null && !this.queryTypes.isEmpty() && !this.queryTypes.contains(queryType)) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isVisible(String user, String role, String sourceIp) {
+        if (this.user != null && !this.user.equals(user)) {
+            return false;
+        }
+        if (this.role != null && !this.role.equals(role)) {
+            return false;
+        }
+        if (this.sourceIp != null) {
+            SubnetUtils.SubnetInfo subnetInfo = new SubnetUtils(sourceIp).getInfo();
+            if (!subnetInfo.isInRange(sourceIp)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public double weight() {
+        double w = 0;
+        if (user != null) {
+            w += 1;
+        }
+        if (role != null) {
+            w += 1;
+        }
+        if (queryTypes != null && !queryTypes.isEmpty()) {
+            w += 1 + 0.1 / queryTypes.size();
+        }
+        if (sourceIp != null) {
+            SubnetUtils.SubnetInfo subnetInfo = new SubnetUtils(sourceIp).getInfo();
+            w += 1 + (Long.numberOfLeadingZeros(subnetInfo.getAddressCountLong() + 2) - 32) / 64.0;
+        }
+        return w;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder classifiersStr = new StringBuilder();
+        classifiersStr.append("(");
+        classifiersStr.append("id=" + id);
+        classifiersStr.append(", weight=" + weight());
+        if (user != null) {
+            classifiersStr.append(", user=" + user);
+        }
+        if (role != null) {
+            classifiersStr.append(", role=" + role);
+        }
+        if (queryTypes != null && !queryTypes.isEmpty()) {
+            List<String> queryTypeList = queryTypes.stream().
+                    map(WorkGroupClassifier.QueryType::name).collect(Collectors.toList());
+            queryTypeList.sort(String::compareTo);
+            String queryTypesStr = String.join(", ", queryTypeList);
+            classifiersStr.append(", query_type in (" + queryTypesStr + ")");
+        }
+        if (sourceIp != null) {
+            classifiersStr.append(", source_ip=" + sourceIp);
+        }
+        classifiersStr.append(")");
+        return classifiersStr.toString();
+    }
+
+    public enum QueryType {
```

Comment:
Why not StmtType?

Reply:
QueryTypes are defined by product manager.